### DISCUSSION TITLE
Building ufs-weather-model prototype-8c fails on frontera

### DISCUSSION THREAD

**Discussion by benjamin-cash** [JCSDA/spack-stack#273] [Good to know]
In the spirit of not having things in emails (I sent this to Dom earlier), I wanted to put this information down here. Using the spack-stack installation @climbfuji  created on frontera, I am attempting to build the Prototype-8c tag of the ufs-weather-model. I am doing the following:

module purge
module use /work2/06146/tg854455/frontera/spack-stack/modulefiles
module load miniconda/3.9.12
module load ecflow/5.8.4

ulimit -s unlimited
module use /work2/06146/tg854455/frontera/spack-stack/spack-stack-v1/envs/skylab-1.0.0-gnu-9.1.0/install/modulefiles/Core
module load stack-gcc/9.1.0
module load stack-mvapich2/2.3
module load stack-python/3.9.12
module load ufs-weather-model-env

CMAKE_FLAGS="-DAPP=S2SWA" CCPP_SUITES="FV3_GFS_v17_coupled_p8" ./build.sh

When I do so, I hit the following errors:
CMake Error at CMakeLists.txt:120 (add_library):
  add_library cannot create ALIAS target "fms" because another target with
  the same name already exists.

CMake Error at /work2/06146/tg854455/frontera/spack-stack/spack-stack-v1/envs/skylab-1.0.0-gnu-9.1.0/install/gcc/9.1.0/cmake-3.22.1-mppepwa/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PIO (missing: C Fortran)
Call Stack (most recent call first):
  /work2/06146/tg854455/frontera/spack-stack/spack-stack-v1/envs/skylab-1.0.0-gnu-9.1.0/install/gcc/9.1.0/cmake-3.22.1-mppepwa/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  CMakeModules/Modules/FindPIO.cmake:184 (find_package_handle_standard_args)
  CMakeLists.txt:124 (find_package)

The modules loaded at this point are:
login2.frontera(503)$ module list

Currently Loaded Modules:
  1) miniconda/3.9.12      9) bacio/2.4.1        17) parallel-netcdf/1.12.2  25) crtm/2.3.0         33) sp/2.3.3                              41) py-numpy/1.22.3      49) py-netcdf4/1.5.3          57) ufs-pyenv/1.0.0
  2) qt5/5.14.2           10) cmake/3.22.1       18) netcdf-c/4.8.1          26) esmf/8.3.0b09      34) ip/3.3.3                              42) py-cftime/1.0.3.4    50) py-bottleneck/1.3.2       58) w3nco/2.4.1
  3) ecflow/5.8.4         11) curl/7.29.0        19) nccmp/1.9.0.1           27) fms/release-jcsda  35) ecbuild/3.6.5                         43) py-cython/0.29.30    51) py-numexpr/2.7.3          59) ufs-weather-model-env/1.0.0
  4) gcc/9.1.0            12) git/2.24.1         20) netcdf-fortran/4.5.4    28) libjpeg/2.1.0      36) gftl-shared/1.5.0                     44) py-f90nml/1.4.2      52) py-six/1.16.0
  5) stack-gcc/9.1.0      13) mvapich2/2.3       21) parallelio/2.5.4        29) jasper/2.0.32      37) yafyaml/0.5.1                         45) py-mpi4py/3.1.2      53) py-python-dateutil/2.8.2
  6) mvapich2-x/2.3       14) pkg-config/0.27.1  22) wget/1.14               30) libpng/1.6.37      38) mapl/2.22.0-esmf-8.3.0b09-esmf-8.3.0  46) py-h5py/3.6.0        54) py-pytz/2021.3
  7) stack-mvapich2/2.3   15) zlib/1.2.12        23) base-env/1.0.0          31) g2/3.4.5           39) openblas/0.3.19                       47) py-markupsafe/2.0.1  55) py-pandas/1.4.0
  8) stack-python/3.9.12  16) hdf5/1.12.1        24) git-lfs/2.10.0          32) g2tmpl/1.10.0      40) py-setuptools/59.4.0                  48) py-jinja2/3.0.3      56) py-pyyaml/6.0

Explicitly loading fms/2022.01 does not solve the problem.

**Comment by climbfuji** [comment]
I think we got past all these errors by now, correct? If so, can you post the correct instructions here?

**Reply by benjamin-cash** [reply]
Yes, I think the issues are no longer related to spack-stack. Here is what I did to build P8.
1 Check out P8 tag
2 Update CMakeModules to develop
3 Change "STATIC" to "SHARED" on line 130 of CMakeLists.txt
4 Add esmf to DEPENDENCIES on line 9 of GOCART/ESMF/Shared/CMakeLists.txt
5 mkdir build && cd build
6 cmake -DAPP=S2SWA -DDEBUG=ON -DCCPP_SUITES=FV3_GFS_v17_coupled_p8 -DINLINE_POST=ON .. 2>&1 | tee log.make
7 Delete -lESMF from CMakeFiles/ufs_model.dir/link.txt
8 make VERBOSE=1 -j4 2>&1 | tee log.make

**Reply by climbfuji** [reply]
Thanks!
